### PR TITLE
[5.5] Add patterns to routeIs method

### PR DESCRIPTION
**Pull request by Lloople**
Same behaviour as `fullUrlIs`. We can check for a group named routes following conventions. For example on a sidebar, checking for `users.*` we can set active to the dropdown when we are on `users.edit`, `users.index`, etc.

Not sure if this should point to 5.4. I can do another pull request if needed

Thanks to @ConnorVG for helping with this :)

### CODE REVIEW

File: src/Illuminate/Http/Request.php [hunk 1]
```diff
@@ -193,14 +193,23 @@ public function is()
...
+     * Determine if the route name matches a pattern.
      *
-     * @param  string  $name
      * @return bool
      */
-    public function routeIs($name)
+    public function routeIs()
     {
-        return $this->route() && $this->route()->named($name);
+        if ((! $route = $this->route()) || ! $routeName = $route->getName()) {
```
Comment (lucasmichot) [inline src/Illuminate/Http/Request.php]: Useless extra brackets; `if (! $route = $this->route() || ! $routeName = $route->getName()) {` is just fine

Comment (Lloople) [inline src/Illuminate/Http/Request.php]: Without that parenthesis around the assignment of `$route`, the IDE is showing the next use of `$route` as an error on the assignment of `$routeName`

Comment (laurencei) [inline src/Illuminate/Http/Request.php]: You never use `$route` - so you dont even need that in the first place?

Should be:

`if (! $this->route() || ! $routeName = $route->getName()) {`

Comment (salomoni) [inline src/Illuminate/Http/Request.php]: Working fine with PHP 7.0.12 without the extra brackets

Comment (Lloople) [inline src/Illuminate/Http/Request.php]: @salomoni Tried without brackets, not passing tests. Adding brackets again

Comment (Lloople) [inline src/Illuminate/Http/Request.php]: @lucasmichot is there any action I need to do in order to get this PR checked? First time PRing here and don't know if I'm missing something

**Comment by ConnorVG** [comment]
Thanks for the credit 😉

**Comment by taylorotwell** [comment]
If there are added here do they need to be added to the route `named` method?

**Comment by Lloople** [comment]
@taylorotwell since we can access the route through different classes or helpers, I think that this logic should move to the `named` method as you said, and call that method in `routeIs`. I can do the change tomorrow afternoon

**Comment by taylorotwell** [comment]
Updated all calls to use argument unpacking and variadic arguments.

**Pull request merged**